modules/telemetry/events/settings-events.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace VIPWorkflow\Modules\Telemetry\Events;
+
+use VIPWorkflow\Modules\Telemetry\Tracker;
+use VIPWorkflow\VIP_Workflow;
+
+function record_admin_update( string $previous_version, string $new_version, Tracker $tracker ): void {
+	// Get all custom statuses
+	$custom_statuses = VIP_Workflow::instance()->custom_status->get_custom_statuses();
+
+	// Get all posts count
+	$posts_count = wp_count_posts();
+	// Only care about published and posts with custom status
+	$total_posts = (int) $posts_count->publish;
+	foreach ( $custom_statuses as $status ) {
+		$total_posts += (int) $posts_count->{ $status->slug };
+	}
+
+	$tracker->record_event( 'administration_update', [
+		'previous_version' => $previous_version,
+		'new_version'      => $new_version,
+		'custom_statuses'  => count( $custom_statuses ),
+		'total_posts'      => $total_posts,
+	] );
+}
+
+function record_settings_update( array $new_options, array $old_options, Tracker $tracker ): void {
+	if ( $new_options['publish_guard'] !== $old_options['publish_guard'] ) {
+		record_publish_guard_toggle( $new_options['publish_guard'], $tracker );
+	}
+
+	if ( $new_options['send_to_webhook'] !== $old_options['send_to_webhook'] ) {
+		record_send_to_webhook_toggle( $new_options['send_to_webhook'], $new_options['webhook_url'], $tracker );
```

Comment:
`webhook_url` may contain private information like an auth key, right? We should probably remove this from the event data.

Reply:
Removed in 82596947dc3c7aa86b53b8b109219e48862fad47